OCR/frontend/src/componets/ExtractUploadFile.tsx

```diff
@@ -10,15 +10,67 @@ interface IFilesObj {
   files: File[];
 }
 
+interface Template {
+  name: string;
+  description: string;
+  pages: {
+    image: string;
+    fieldNames: string[];
+  }[];
+}
+
 export const ExtractUploadFile: React.FC<ExtractUploadFileProps> = ({
   onUploadComplete,
 }) => {
   const id = useId();
   const { addFile } = useFiles();
   const [template, setTemplate] = useState<string>("");
   const [uploadProgress, setUploadProgress] = useState<number>(0);
+  const [templates, setTemplates] = useState<Template[]>([]);
   const [uploadedFile, setUploadedFile] = useState<File | null>(null);
 
+  // Load templates from local storage, and if none are found, load test data
+  const loadTemplatesFromLocalStorage = () => {
+    const storedTemplates = localStorage.getItem("templates");
+    if (storedTemplates) {
+      const parsedTemplates = JSON.parse(storedTemplates);
+      setTemplates(parsedTemplates);
+    } else {
+      loadTemplatesFromLocalStorageTestData();
+    }
+  };
+
+  const loadTemplatesFromLocalStorageTestData = () => {
```

Comment:
nitpick, rename this to something more accurate, like `loadTemplatesTestData` since it has nothing to do with LocalStorage